### ISSUE TITLE
DATA Race on client close

### ISSUE THREAD

**Issue by kozlovic** [nats-io/nats-streaming-server#113]
```
==================
WARNING: DATA RACE
Read by goroutine 15:
  github.com/nats-io/nats-streaming-server/server.(*subState).deleteFromList()
      /home/travis/gopath/src/github.com/nats-io/nats-streaming-server/server/server.go:1551 +0x7f
  github.com/nats-io/nats-streaming-server/server.(*subStore).Remove()
      /home/travis/gopath/src/github.com/nats-io/nats-streaming-server/server/server.go:309 +0x3a3
  github.com/nats-io/nats-streaming-server/server.(*StanServer).removeAllNonDurableSubscribers()
      /home/travis/gopath/src/github.com/nats-io/nats-streaming-server/server/server.go:1596 +0x270
  github.com/nats-io/nats-streaming-server/server.(*StanServer).closeClient()
      /home/travis/gopath/src/github.com/nats-io/nats-streaming-server/server/server.go:1140 +0x1bb
  github.com/nats-io/nats-streaming-server/server.(*StanServer).processCloseRequest()
      /home/travis/gopath/src/github.com/nats-io/nats-streaming-server/server/server.go:1156 +0x302
  github.com/nats-io/nats-streaming-server/server.(*StanServer).(github.com/nats-io/nats-streaming-server/server.processCloseRequest)-fm()
      /home/travis/gopath/src/github.com/nats-io/nats-streaming-server/server/server.go:913 +0x37
  github.com/nats-io/nats.(*Conn).waitForMsgs()
      /home/travis/gopath/src/github.com/nats-io/nats/nats.go:1360 +0x3aa
Previous write by goroutine 24:
  runtime.slicecopy()
      /home/travis/.gimme/versions/go1.6.2.linux.amd64/src/runtime/slice.go:113 +0x0
  github.com/nats-io/nats-streaming-server/server.findBestQueueSub()
      /home/travis/gopath/src/github.com/nats-io/nats-streaming-server/server/server.go:1251 +0x1be
  github.com/nats-io/nats-streaming-server/server.(*StanServer).sendMsgToQueueGroup()
      /home/travis/gopath/src/github.com/nats-io/nats-streaming-server/server/server.go:1264 +0x87
  github.com/nats-io/nats-streaming-server/server.(*StanServer).performAckExpirationRedelivery()
      /home/travis/gopath/src/github.com/nats-io/nats-streaming-server/server/server.go:1425 +0xd45
  github.com/nats-io/nats-streaming-server/server.(*StanServer).setupAckTimer.func1()
      /home/travis/gopath/src/github.com/nats-io/nats-streaming-server/server/server.go:1522 +0x36
Goroutine 15 (running) created at:
  github.com/nats-io/nats.(*Conn).subscribe()
      /home/travis/gopath/src/github.com/nats-io/nats/nats.go:1828 +0x515
  github.com/nats-io/nats.(*Conn).Subscribe()
      /home/travis/gopath/src/github.com/nats-io/nats/nats.go:1745 +0x7a
  github.com/nats-io/nats-streaming-server/server.(*StanServer).initSubscriptions()
      /home/travis/gopath/src/github.com/nats-io/nats-streaming-server/server/server.go:913 +0xb17
  github.com/nats-io/nats-streaming-server/server.RunServerWithOpts()
      /home/travis/gopath/src/github.com/nats-io/nats-streaming-server/server/server.go:575 +0x106b
  github.com/nats-io/nats-streaming-server/server.testStalledRedelivery()
      /home/travis/gopath/src/github.com/nats-io/nats-streaming-server/server/server_test.go:1065 +0x1713
  github.com/nats-io/nats-streaming-server/server.TestStalledQueueRedelivery()
      /home/travis/gopath/src/github.com/nats-io/nats-streaming-server/server/server_test.go:1082 +0x43
  testing.tRunner()
      /home/travis/.gimme/versions/go1.6.2.linux.amd64/src/testing/testing.go:473 +0xdc
Goroutine 24 (running) created at:
  time.goFunc()
      /home/travis/.gimme/versions/go1.6.2.linux.amd64/src/time/sleep.go:129 +0x6d
==================
```

**Comment by derekcollison** [comment]
This in server or NATS client?

**Comment by kozlovic** [comment]
NATS Streaming Server, and a PR is already submitted.

**Comment by kozlovic** [comment]
Resolved by #114
